plugins/workspace-plugin/tests/workspace-projects-manager.spec.ts

```diff
@@ -257,73 +256,6 @@ describe('Test Workspace Projects Manager', () => {
     expect(showInfoMessageMock).toBeCalledTimes(0);
   });
 
-  test('Should create a git project in a Devfile', async () => {
-    getDevfileMock.mockReturnValue(devfile_Without_Attributes);
-
-    const getUpstreamBranchSpy = jest.spyOn(git, 'getUpstreamBranch');
-    getUpstreamBranchSpy.mockResolvedValueOnce({
-      branch: 'branch',
-      remote: 'remote',
-      remoteURL: 'someRemoteUrl',
-    });
-
-    await workspaceProjectsManager.updateOrCreateProjectInWorkspace('/projects/che-theia');
-
-    expect(updateOrCreateGitProjectInDevfileMock).toBeCalledTimes(1);
-  });
-
-  test('Should NOT create a project when given Uri is not defined', async () => {
-    await workspaceProjectsManager.updateOrCreateProjectInWorkspace('');
-
-    expect(getDevfileMock).toBeCalledTimes(0);
-    expect(updateDevfileMock).toBeCalledTimes(0);
-    expect(updateOrCreateGitProjectInDevfileMock).toBeCalledTimes(0);
-  });
-
-  test('Should NOT create a git project in a Devfile when `remoteURL` is not defined', async () => {
-    getDevfileMock.mockReturnValue(devfile_Without_Attributes);
-
-    const getUpstreamBranchSpy = jest.spyOn(git, 'getUpstreamBranch');
-    getUpstreamBranchSpy.mockResolvedValueOnce({
-      branch: 'branch',
-      remote: 'remote',
-      remoteURL: '', // not defined
-    });
-
-    await workspaceProjectsManager.updateOrCreateProjectInWorkspace('/projects/che-theia');
-
-    expect(updateOrCreateGitProjectInDevfileMock).toBeCalledTimes(0);
-  });
-
-  test('Should NOT create a git project in a Devfile when upstream branch is `undefined`', async () => {
-    getDevfileMock.mockReturnValue(devfile_Without_Attributes);
-
-    const getUpstreamBranchSpy = jest.spyOn(git, 'getUpstreamBranch');
-    getUpstreamBranchSpy.mockResolvedValueOnce(undefined);
-
-    await workspaceProjectsManager.updateOrCreateProjectInWorkspace('/projects/che-theia');
-
-    expect(updateOrCreateGitProjectInDevfileMock).toBeCalledTimes(0);
-  });
-
-  test('Should delete a project from Devfile', async () => {
-    getDevfileMock.mockReturnValue(devfile_Without_Attributes);
-
-    await workspaceProjectsManager.deleteProjectInWorkspace('/projects/che-theia');
-
-    expect(getDevfileMock).toBeCalledTimes(1);
-    expect(updateDevfileMock).toBeCalledTimes(1);
-    expect(deleteProjectFromDevfileMock).toBeCalledTimes(1);
-  });
-
-  test('Should NOT delete a project when given Uri is not defined', async () => {
-    await workspaceProjectsManager.deleteProjectInWorkspace('');
-
-    expect(getDevfileMock).toBeCalledTimes(0);
-    expect(updateDevfileMock).toBeCalledTimes(0);
-    expect(deleteProjectFromDevfileMock).toBeCalledTimes(0);
-  });
-
   // Backward compatibility for single-root workspaces
   // we need it to support workspaces which were created before switching multi-root mode to ON by default
   test('Should add projects as workspace folders when projects already exist on file system', async () => {
```

Comment:
Do you have an idea why in the tests report backward compatibility use case is displayed as NOT covered after your changes?
I mean - it looks like you didn't remove the corresponding tests, but it's displayed as uncovered, so maybe something wrong in the provided logic or something else... 

In general I see a regression in tests coverage:
- for `workspace-projects-manager.ts` it was 98.21% lines and with your changes it's 79.57
- for `workspace-folder-updater.ts` it was 100 % lines ===> 92.19

Before:
![before](https://user-images.githubusercontent.com/5676062/116353570-ad3eb880-a7ff-11eb-955f-28f4365bb430.png)

After:
![after](https://user-images.githubusercontent.com/5676062/116353597-b891e400-a7ff-11eb-856c-2c4fe96fe299.png)

Reply:
I added some code to that files and looks like haven't written tests for some pieces of code. I will improve it right now.